assets/js/util/test/numberFormat.js

```diff
@@ -115,5 +116,105 @@ describe( 'numberFormat', () => {
 	it.each( siteKitLocales )( 'formats numbers correctly with locale variant %s', ( locale, value, expected ) => {
 		setupGoogleSiteKit( locale );
 		expect( numberFormat( value ) ).toStrictEqual( expected );
+		expect( console ).not.toHaveWarned();
+	} );
+
+	describe( 'graceful degradation for problematic options in some browsers', () => {
+		let NumberFormatSpy;
+
+		afterEach( () => {
+			if ( NumberFormatSpy.mockRestore ) {
+				NumberFormatSpy.mockRestore();
+			}
+		} );
+
+		// Error message that browser throws on error.
+		const errorMessage = 'TypeError: Failed to initialize NumberFormat since used feature is not supported in the linked ICU version';
+
+		/**
+		 * This function mocks the implementation of certain errors identified in
+		 * https://github.com/google/site-kit-wp/issues/3255
+		 * so that we can test that we've properly handled them.
+		 */
+		const mockNumberFormat = () => {
+			const NumberFormat = Intl.NumberFormat;
+			NumberFormatSpy = jest.spyOn( global.Intl, 'NumberFormat' );
+			NumberFormatSpy.mockImplementation( ( locales, options ) => {
+				if ( undefined === options ) {
+					return NumberFormat( locales, options );
+				}
+				const optionsThatError = {
+					currencyDisplay: 'narrow',
+					currencySign: 'accounting',
+					style: 'unit',
+					signDisplay: '*', // * denotes any value.
+					compactDisplay: '*',
+				};
+				for ( const [ key, value ] of Object.entries( options ) ) {
+					if ( optionsThatError[ key ] && [ value, '*' ].includes( optionsThatError[ key ] ) ) {
+						throw new TypeError( errorMessage );
+					}
+				}
```

Comment:
Maybe we can simplify this a bit while improving readability of the tests by introducing a little helper here like `createThrowIfOptionMatch( key, [ value ] )` which would return a function that throws if the key and optionally the value matches, otherwise forwarding through to the original implementation.

E.g.
```js
function createThrowIfOptionMatch( key, value ) {
    return ( locale, options = {} ) => {
        if ( options.hasOwnProperty( key ) ) {
            if ( value === options[ key ] || value === undefined ) {
                throw new Error( errorMessage );
            }
        }
        return originalNumberFormat( locale, options );
    }
}
```

Then in the test we could call e.g. `NumberFormatSpy.mockImplementation( createThrowIfOptionMatch( 'currencyDisplay', 'narrow' ) )` rather than `mockNumberFormat`.

What do you think?

Reply:
Updated in https://github.com/google/site-kit-wp/pull/3620/commits/a2cabed123c702969fffad9ed2e122f7d9edc89c